ir/translation/literal.rs

```diff
@@ -217,9 +217,15 @@ impl FromTypeQLLiteral for TimeZone {
                     .map_err(|_| LiteralParseError::InvalidTimezoneNamed { name: name.clone(), source_span })?,
             )),
             typeql::value::TimeZone::ISO(value) => {
-                Ok(TimeZone::Fixed(FixedOffset::from_str(value).map_err(|_| {
-                    LiteralParseError::InvalidTimezoneFixedOffset { offset: value.clone(), source_span }
-                })?))
+                if value.ends_with("z") || value.ends_with("Z") {
```

Comment:
Lowercase `z` is not a valid ISO 8601 timezone specifier. It's not accepted by TypeQL either.